analyzers/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -97,5 +106,8 @@ private void CheckConditional(SyntaxNodeAnalysisContext context)
             }
             CheckTernaryExpressionBranches(context, conditional.SyntaxTree, whenTrue, whenFalse);
         }
+
+        private static bool IsThrowExpression(ExpressionSyntax expressionSyntax) =>
+            ThrowExpressionSyntaxWrapper.IsInstance(expressionSyntax.RemoveParentheses());
```

Comment:
in practice I think `RemoveParentheses()` is overkill and wasting perf , so I'd remove it

Reply:
In this case I agree to remove it, just document the case in the UTs to know that we have FNs.
In general I would say that  `RemoveParentheses()` is not always an overkill, specially in rules which handle expressions with logical operators.

Reply:
totally agree with you for the general usecase